app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt

```diff
@@ -118,6 +119,40 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
         }
     }
 
+    private fun setReportButton() {
+        //Only enable the feature when the user is logged in
+        if (PrefManager.getToken() != null) {
+            reportSkill.visibility = View.VISIBLE
+            reportSkill.setOnClickListener {
+                val dialogBuilder = AlertDialog.Builder(requireContext())
+                val view = layoutInflater.inflate(R.layout.alert_report_skill, null)
+                val reportMessage = view.findViewById(R.id.reportMessage) as EditText
+                dialogBuilder.setView(view)
+                dialogBuilder.setTitle(R.string.report_skill)
+
+                dialogBuilder.setPositiveButton(R.string.report_send) { dialog, whichButton ->
+                    if (PrefManager.getToken() != null && reportMessage.text.isNotEmpty()) {
+                        val map: MutableMap<String, String> = HashMap()
+                        map.put("access_token", PrefManager.getToken().toString())
+                        map.put("feedback", reportMessage.text.toString())
+                        map.put("skill", skillTag)
+                        map.put("group", skillData.group)
+                        map.put("model", skillData.model)
+                        skillDetailsPresenter.sendReport(map)
```

Comment:
use objects. You may use a data class for the same. Pass the object rather than passing map as param. Then, pass map to the final method, defined in SusiService.java, using ClientBuilder.java. Please check the implementation for other features like feedback for a reference.

Reply:
@arundhati24 What effect will it have? I have just implemented the same way all other functionalities are implemented.

Reply:
@arundhati24 In the implementation of feedback, we have already extracted the same things, then why are we sending another object and do the same thing again?

In SkillDetailsModel.

```
 val query: MutableMap<String, String> = HashMap()
        query.put("model", queryObject.model)
        query.put("group", queryObject.group)
        query.put("language", queryObject.language)
        query.put("skill", queryObject.skill)
        query.put("feedback", queryObject.feedback)
        query.put("access_token", queryObject.accessToken)
```

same fields again extracted.

Reply:
You can refer to #1491 and also PRs related to feedback features as this has been discussed a lot already. We don't want to use maps directly as it is not safe. Also, as our mentor  @iamareebjamal has already mentioned - not to use maps in views.

Reply:
@arundhati24 Sure, moving the code and sending the PR asap.!